apps/client/components/Dialog/AcademicDialog.tsx

```diff
@@ -111,92 +107,133 @@ export const AcademicDialog: FC<Props> = ({handleClose, OnSubmit, isOpen}) => {
           <Formik
             initialValues={initialValues}
             validationSchema={validationSchema}
-            onSubmit={(values) => OnSubmit({
-              ...values,
-              city: academicDetail.city,
-              age: date.format('YYYY-MM-DD'),
-              subject: academicDetail?.subject,
-              degree: academicDetail.degree,
-              gender: academicDetail.gender,
-              isAgree: checked,
-              imageUrl: imageUrl,
-              company: academicDetail?.company,
-              career: academicDetail?.career
-            })}
+            onSubmit={async (values) => {
+              if (!loading) {
+                setLoading(true);
+                const res = await OnSubmit({
+                  ...values,
+                })
+                setData(res)
+                timer.current = window.setTimeout(() => {
+                  setLoading(false);
+                })
+              }
+            }}
```

Comment:
please change this function to be outside the components

Reply:
Done